### PR TITLE
Add types for tokenizr

### PR DESCRIPTION
**Pull request by OldStarchy**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/tokenizr/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
+// Type definitions for tokenizr 1.5
+// Project: https://github.com/rse/tokenizr
+// Definitions by: Nicholas Sorokin <https://github.com/aNickzz>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export {};
+
+export default class Tokenizr {
```
Comment (plantain-00) [inline types/tokenizr/index.d.ts]: It's not `export default`: https://github.com/rse/tokenizr/blob/master/src/tokenizr.js#L713

File: types/tokenizr/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,264 @@
+// Type definitions for tokenizr 1.5
+// Project: https://github.com/rse/tokenizr
+// Definitions by: Nicholas Sorokin <https://github.com/aNickzz>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export {};
```
Comment (plantain-00) [inline types/tokenizr/index.d.ts]: I think this line can be removed

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/tokenizr/index.d.ts (hunk 1 above)
Comment (aNickzz) [reply types/tokenizr/index.d.ts]: I've removed "default" however I'm not sure of the correct way to declare the export. With the "export default" I was able to build with these types in another project (I'll test this again tonight with the changes I've just comitted)

File: types/tokenizr/index.d.ts (hunk 2 above)
Comment (aNickzz) [reply types/tokenizr/index.d.ts]: I added this as a recommendation from the linter as `Action` and `RuleAction` are not explicitly (or implicitly) exported.

>All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting.

Those types are not part of tokenizrs exports, I just didn't want to copy/paste the definitions in the places its used.

File: types/tokenizr/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+export class Tokenizr {
```
Comment (plantain-00) [inline types/tokenizr/index.d.ts]: It should be like this:

```ts
class Tokenizr { ... }

export = Tokenizr
```

```ts
import Tokenizr = require('tokenizr');

const lexer = new Tokenizr();
```

https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Review by plantain-00 — APPROVED** [review]

File: types/tokenizr/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+    isA(type: string, value?: any): boolean;
+
+    /**
+     * Render a useful string representation
+     */
+    toString(): string;
+}
+
+export = Tokenizr;
```
Comment (gabritto) [inline types/tokenizr/index.d.ts]: this looks good, but i noticed the original library [exposes some of those classes as properties of Tokenizr](https://github.com/rse/tokenizr/blob/master/src/tokenizr.js#L708). how are you exporting those properties?

Comment (aNickzz) [reply types/tokenizr/index.d.ts]: They are declared on the Tokenizr class, lines 139 - 141.

Comment (gabritto) [reply types/tokenizr/index.d.ts]: ok, i see them now. thanks!

**Pull request merged**